### PR TITLE
5490 sponsor information carry

### PR DESCRIPTION
**Pull request by cvalarida**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1924

## No Previous Benefits
image

## Previous Benefits
image

image

### CODE REVIEW

File: src/js/edu-benefits/5490/config/form.js [hunk 1]
```diff
@@ -361,32 +361,52 @@ const formConfig = {
...
+                hideIf: (formData) => formData['view:currentSameAsPrevious']
               },
-              vaFileNumber: {
-                'ui:title': 'Sponsor file number',
-              }
-            }),
+              veteranFullName: _.merge(fullNameUi, {
+                'ui:options': {
+                  updateSchema: (data, form) => {
+                    if (!_.get('sponsorInformation.data.view:currentSameAsPrevious', form)) {
```
Comment (raquelromano-gov) [inline src/js/edu-benefits/5490/config/form.js]: I know this was already here, but I find it a bit more readable to test
```
if (_.get('sponsorInformation.data.view:currentSameAsPrevious', form)) {
  return nonRequiredFullName;
}
return fullName;
```

File: src/js/edu-benefits/5490/helpers.jsx [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
-  const formData = transformForSubmit(formConfig, form);
+  // Clone the form in so we don't modify the original...because of reasons FP
+  const newForm = _.cloneDeep(form);
+
+  // Copy the data if necessary
+  if (form.sponsorInformation.data['view:currentSameAsPrevious']) {
+    newForm.sponsorInformation.data['view:currentSponsorInformation'] = {
+      veteranFullName: form.benefitHistory.data.previousBenefits.veteranFullName,
+      'view:veteranId': {
+        veteranSocialSecurityNumber: form.benefitHistory.data.previousBenefits.veteranSocialSecurityNumber
```
Comment (raquelromano-gov) [inline src/js/edu-benefits/5490/helpers.jsx]: I'm now surprised that for previouesBenefits, we're not allowing either the SSN or VA File Number, but that's another story for another day/PR. (I think we should ask stakeholders about that--would be a better experience to be consistent about what info is required about any sponsor.)

**Review by raquelromano-gov — APPROVED** [review]

**Comment by goldenmeanie** [comment]
Is the backend duplicating the info from the previous benefits in the appropriate place in the spool file?

**Comment by raquelromano-gov** [comment]
It should be--Chris updated the function that transforms the data before stringifying the JSON to send to the api.

**Comment by cvalarida** [comment]
@goldenmeanie Yes it is; I did test it manually to confirm that if that checkbox is checked, the information is carried over from the previous benefits sponsor to the current benefits sponsor.

@raquelromano-gov An excellent suggestion! I think we definitely should. I'll add it in here.

**Comment by cvalarida** [comment]
@raquelromano-gov Can you review again? I just added the test for verifying the data is what we expect it to be (thereby adding an additional bit of functionality to this particular schema test file).

File: src/js/edu-benefits/5490/helpers.jsx (hunk 2 above)
Comment (raquelromano-gov) [inline src/js/edu-benefits/5490/helpers.jsx]: Verified this is a good idea--separate ticket filed and added to current sprint: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1929

**Review by raquelromano-gov — APPROVED** [review]
LGTM

File: test/edu-benefits/5490/schema/schema.unit.spec.js [hunk 3]
```diff
@@ -26,5 +27,13 @@ describe('5490 schema tests', () => {
         }
         expect(result.valid).to.be.true;
       });
+      // Expect the data to be what's found in expected-data/${file}
+      if (expectedDataFiles.indexOf(file) !== -1) {
```
Comment (raquelromano-gov) [inline test/edu-benefits/5490/schema/schema.unit.spec.js]: Should probably add a comment here to replace this with includes() once the version of Node.js on Jenkins is upgraded

**Pull request merged**